lib/PrintAsClang/PrintClangValueType.cpp

```diff
@@ -263,12 +263,12 @@ void ClangValueTypePrinter::printValueTypeDecl(
         "*>(other._getOpaquePointer()), metadata._0);\n";
   os << "  }\n";
 
-  // FIXME: the move constructor should be hidden somehow.
-  os << "  inline ";
+  // FIXME: implement the move constructor.
+  os << "  [[noreturn]] inline ";
   printer.printBaseName(typeDecl);
   os << "(";
   printer.printBaseName(typeDecl);
-  os << " &&) = default;\n";
+  os << " &&) { abort(); }\n";
```

Comment:
Could we emit `= delete` instead? That might be more helpful for the user since accidentally using the move constructor would be a compile error instead of a runtime error.

Reply:
I can't emit delete, as C++ will not compile the elided moves that are not actually runtime moves, that I need to return such type from the thunk wrapper .

Reply:
```
inline SwiftType thunk() {
  auto result = SwiftType::make();
  $swiftFunction(&result);
  return result; // <-- this will not compile if the move constructor is deleted, but it's elided according to C++ rules so it does not trap.
}
```

Reply:
I see, makes sense, thanks.